cmd/pilot-agent/main.go

```diff
@@ -63,8 +66,10 @@ var (
 			glog.V(2).Infof("sidecar %#v", sidecar)
 			glog.V(2).Infof("mesh configuration %#v", mesh)
 
-			if err = os.MkdirAll(configpath, 0700); err != nil {
-				return multierror.Prefix(err, "failed to create directory for proxy configuration")
+			if len(customConfig) == 0 {
+				if err = os.MkdirAll(configpath, 0700); err != nil {
```

Comment:
Will update - but that part of the code MUST be fixed, we can't fetch ingress TLS private keys over http with no authentication.